# 🎗️ Confidence Interval
A confidence interval is a range of values,calculated from sample data that is likely to contain the true population parameter with a specific probability (e.g., 95% or 99%)

$\bar{X} \; \pm \; Z \cdot \frac{\sigma}{\sqrt{n}}$

- $\bar{X}$ = sample mean  
- $Z$ = Z-score for the desired confidence level (e.g., 1.96 for 95%)  
- $\sigma$ = population standard deviation  
- $n$ = sample size

---

#### » Import Numpy and create random integer array of 1000  to simulate prices ranges 10 and 110 given 1000 people

In [1]:
import numpy as np
prices = np.random.randint(10,110,1000)
prices

array([ 66,  73,  54,  62,  37,  24,  20,  56,  75,  81,  23,  17, 101,
        13,  58,  89,  60,  99,  74,  80,  54,  60,  94,  80,  52,  90,
        98,  32,  84,  72,  73,  95,  56,  32,  48,  49,  68,  36,  90,
        96,  35, 107,  77,  38,  34,  16,  69,  20,  87,  41,  18,  42,
       106,  95,  33,  90,  74,  82,  80, 103,  41,  91,  63,  56,  74,
        66,  54,  48,  80, 109,  98,  52,  50,  65,  38,  37,  41, 109,
        62,  83,  92,  79,  80,  93,  32,  68,  73,  19, 107,  58,  93,
       109, 100,  81,  78,  45,  25,  35,  23,  52,  17,  60,  13,  21,
        60, 103,  13,  79,  42,  25,  37,  48,  36,  70,  59,  75,  90,
        77,  40,  87,  67,  18,  28,  94,  27,  31, 108,  63,  32,  41,
        75,  15,  40,  20,  74,  75,  12,  77,  47,  76,  93,  97,  66,
        83, 103,  28,  48,  79,  77,  66,  96,  49,  51,  38, 101,  57,
        88, 108,  68,  46,  78,  71,  38,  20,  89,  38,  40,  93, 100,
        65,  43,  73,  85,  41, 105,  20,  99,  95,  58,  79,  9

#### » Calculate the mean of the prices array

In [2]:
prices.mean()

np.float64(60.468)

#### » Calculate the %95 reliability price range by importing "statsmodels"

In [5]:
import statsmodels.stats.api as sms
sms.DescrStatsW(prices).tconfint_mean()

(np.float64(58.704608815566786), np.float64(62.23139118443322))

The result shows that if the price is betweeen at 58.7 and 62.23, people are 95% likely to buy it.